src/function/arithmetic/multiply.js

```diff
@@ -756,6 +766,11 @@ export const createMultiply = /* #__PURE__ */ factory(name, dependencies, ({ typ
     // update ptr
     cptr[bcolumns] = cindex.length
 
+    // Set the type to undefined if it wasn't defined for a or b
+    if (adt !== a._datatype || bdt !== b._datatype) {
```

Comment:
Same here, can you move the logic to the place where the matrix is created?

Reply:
Done